packages/typespec-python/src/types.ts

```diff
@@ -480,3 +486,55 @@ export function emitEndpointType<TServiceOperation extends SdkServiceOperation>(
     }
     return params;
 }
+
+function getXmlMetadata(type: SdkType | SdkModelPropertyType): Record<string, any> {
+    const xmlMetadata: Record<string, any> = {};
+    const xmlDecorators = type.decorators.filter(
+        (x) => x.name.startsWith("TypeSpec.Xml.") || x.name.startsWith("TypeSpec.@encodedName"),
+    );
+    for (const decorator of xmlDecorators) {
+        switch (decorator.name) {
+            case "TypeSpec.@encodedName":
+                if (decorator.arguments["mimeType"] === "application/xml") {
+                    xmlMetadata["name"] = decorator.arguments["name"];
+                    break;
+                }
+                continue;
+            case "TypeSpec.Xml.@attribute":
+                xmlMetadata["attribute"] = true;
+                break;
+            case "TypeSpec.Xml.@name":
+                xmlMetadata["name"] = decorator.arguments["name"];
+                break;
+            case "TypeSpec.Xml.@ns":
+                if (decorator.arguments["ns"].kind === "enumvalue") {
+                    xmlMetadata["namespace"] = (decorator.arguments["ns"] as SdkEnumValueType).value;
+                    xmlMetadata["prefix"] = (decorator.arguments["ns"] as SdkEnumValueType).name;
+                } else {
+                    xmlMetadata["namespace"] = decorator.arguments["ns"];
+                    xmlMetadata["prefix"] = decorator.arguments["prefix"];
+                }
+                break;
+            case "TypeSpec.Xml.@unwrapped":
+                if (type.kind === "property" && type.type.kind === "array") {
+                    xmlMetadata["unwrapped"] = true;
+                } else {
+                    xmlMetadata["text"] = true;
+                }
+                break;
+        }
+    }
+    // add item metadata for array
+    if (type.kind === "property" && type.type.kind === "array" && type.type.valueType.kind !== "model") {
+        const itemMetadata = getXmlMetadata(type.type.valueType);
+        // if array item is a primitive type, we need to use itemsName to change the name
+        if (Object.keys(itemMetadata).length > 0) {
```

Comment:
nit: should these just correspond 1-1 with what tsp is returning to us? I.e. `xmlMetadata["name"] = itemMetadata["name"]`

Reply:
tsp only returns the user defined decorator data. xml has some convention that i need to specific handle. so the item related metadata is only for ser/deser in python.